types/jest/index.d.ts

```diff
@@ -299,6 +299,7 @@ declare namespace jest {
 
     interface Each {
         // Exclusively arrays.
+        <T extends ReadonlyArray<any>>(cases: ReadonlyArray<T>): (name: string, fn: (...args: [T[0], T[1], T[2], T[3], T[4], T[5], T[6], T[7], T[8], T[9]]) => any, timeout?: number) => void;
```

Comment:
This should probably be:
```suggestion
        <T extends ReadonlyArray<any>>(cases: ReadonlyArray<T>): (name: string, fn: (...args: T) => any, timeout?: number) => void;
```

Reply:
I tried. But TS was complaining about spread args can't be read-only array.

Reply:
This will convert it to a read‑write array.

Also, you might want to move this after `T extends any[]`, since `any[]` extends `readonly any[]`.

```suggestion
        <T extends ReadonlyArray<any>>(cases: ReadonlyArray<T>): (name: string, fn: (...args: T extends ReadonlyArray<infer U> ? U[] : any[]) => any, timeout?: number) => void;
```

Reply:
@ExE-Boss I think that will infer the union of all possible types used in the inner arrays (the cases), but not at the actual argument index (which is more accurate).

Reply:
@ExE-Boss I think that will infer the union of all possible types used in the inner arrays (the cases), but not at the actual argument index when declared as a tuple (which is more accurate).

Reply:
@wsmd I think you are right. 

Any ideas on how should we type this better?

Reply:
@ExE-Boss I have removed the `<T extends any[]>`.

Reply:
@ycmjason

> I have removed the `<T extends any[]>`.

That&nbsp;only&nbsp;makes the&nbsp;situation&nbsp;worse for&nbsp;read&#x2011;write&nbsp;unions of&nbsp;11 or&nbsp;more&nbsp;elements.